packages/suite/src/views/wallet/send/Outputs/TokenSelect/TokenSelect.tsx

```diff
@@ -0,0 +1,420 @@
+                            }))}
+                            isDisabled={options.length === 1}
+                            formatOptionLabel={(option: SelectAssetOptionCurrencyProps) => {
+                                return (
+                                    <Row justifyContent="flex-start" gap={spacings.sm}>
+                                        {selectedOption &&
+                                        selectedOption.contractAddress !== null ? (
+                                            <AssetLogo
+                                                coingeckoId={option.coingeckoId}
+                                                contractAddress={
+                                                    option.contractAddress
+                                                        ? getContractAddressForNetwork(
+                                                              account.symbol,
+                                                              option.contractAddress,
+                                                          )
+                                                        : undefined
+                                                }
+                                                size={24}
+                                                placeholder={option.symbol.toUpperCase()}
+                                                placeholderWithTooltip={false}
+                                                shouldTryToFetch={true}
+                                            />
+                                        ) : (
+                                            <CoinLogo symbol={account.symbol} size={24} />
+                                        )}
+                                        <Column alignItems="flex-start">
+                                            <Row justifyContent="flex-start">
+                                                <Text variant="default" typographyStyle="body">
+                                                    {option.cryptoName}
+                                                </Text>
+                                            </Row>
+                                            <Row>
+                                                <Text variant="tertiary" typographyStyle="hint">
+                                                    <HiddenPlaceholder>
+                                                        {option.balance ?? '0'}
+                                                    </HiddenPlaceholder>{' '}
+                                                    {option.symbol.toUpperCase()}
+                                                    {!isTestnet(
+                                                        option.networkSymbol as NetworkSymbol,
+                                                    ) &&
+                                                        option.balance &&
+                                                        ' ≈ '}
```

Comment:
nit: one ternary would be nicer than the two `&&` short circuits.
:thought_balloon:  Though ideally, the `FiatValue` would also be in the condition. This code relies on the behavior that it returns nothing, which is not so clear. But that's fine
```suggestion
                                                    (!isTestnet(option.networkSymbol as NetworkSymbol) && option.balance) ? ' ≈ ' : null
```

Reply:
Yea, now it renders `≈` even though fiat rate is not available

Reply:
I've added formattedCryptoAmount for the balance above 

```
                                    <HiddenPlaceholder>
                                        <FormattedCryptoAmount
                                            value={selectedOption.balance ?? '0'}
                                            symbol={selectedOption.symbol}
                                        />
                                    </HiddenPlaceholder>
```

Reply:
I can try something like

```ts
const renderFiatValueComponent = () => {
    const fiatValue = (
        <FiatValue
            tokenAddress={option.contractAddress ? (option.contractAddress as TokenAddress) : undefined}
            amount={option.balance ?? '0'}
            symbol={option.networkSymbol}
        />
    );
    
    return fiatValue;
};
```

and use it like this
```
                    {renderFiatValueComponent() && (
                        <> ≈ {renderFiatValueComponent()} </>
                    )}
```

but I am not sure if it would work

Reply:
Oh I get it now, it's more complex than I thought at first, because `FiatValue` can render null if fiat rate is not available, which is unknown here :confused:

Reply:
it'd be ideal to extract the `FiatValue` inner logic to hook, so we can access the logic without rendering the component and checking its return for `null`. But if we have to, I'd render it to a constant:
```
const fiatValue = <FiatValue ...props>;
```
and then use it like
```
{fiatValue ? <> ≈ {fiatValue} </> : null}
```
Which is the same thing as you have written but simpler.

Reply:
I've done it with the logic you mentioned in slack, i will create and issue to revisit this, this is not the only occurence